Automatically start/stop WiFi

c03rad0r opened the pull request.

Commit:
Automatically start/stop WiFi

Commit:
Only setup opennds once. Do I still get a DHCP lease?

Commit:
Does calling setup_opennds twice consecutively fix the dhcp lease problem?

Commit:
Maybe this is good enough for the DHCP lease...

Commit:
Simply restarting didn't get the dhcp lease. Does this script fix it?

chGoodchild commented:
> TODO: Random IP address...

Created a separate issue for this:
https://github.com/OpenTollGate/TollGateNostrToolKit/issues/27

chGoodchild commented:
TODO: also turn off wifi when `ndsctl json` isn't ready yet.

Commit:
Various wifi and DHCP related changes

* Turn off wifi if ndsctl json isn't ready
* use npub instead of MAC
* re-run opennds setup to figure out if DHCP lease is now granted

Commit:
Dedicated dhcp script and debugging for wifi on/off script

Commit:
Added logging line

Commit:
Switched to here_document to fix syntax

Commit:
Enable wifi

chGoodchild commented:
6ef91a73e51525965604dc4999cba54d2a049112

Only the DHCP lease fix needs to be simplified now...

Commit:
Simplified dhcp_lease_fix

Commit:
Reintroduced ipv6 stuff to DHCP_lease fix

Commit:
Reverted dhcp_lease_fix

Commit:
Only removed theme voucher

Commit:
Leaving out opennds restart

Commit:
Reintroduced theme_voucher

Commit:
Doesn't make sense, dhcp lease still not working

Commit:
Added script explanation

chGoodchild commented:
Need to check 28156e074b2a2f498c3b1cffafb434b04a461080 again..

chGoodchild commented:
> Need to check [28156e0](https://github.com/OpenTollGate/TollGateNostrToolKit/commit/28156e074b2a2f498c3b1cffafb434b04a461080) again..

28156e074b2a2f498c3b1cffafb434b04a461080 works...

Commit:
Revert to setup_opennds for resolving dhcp lease

Commit:
Using dhcp fix script instead of opennds

Commit:
DHCP fix

Commit:
dhcp_lease_fix still doesn't work...

I don't understand why repeating setup_opennds works, but dhcp_lease_fix doesn't. What's the difference?

Commit:
Repeat setup opennds

Commit:
Hotspot manager can restart network now

Commit:
Removed hotspot manager to see if dhcp error goes away

chGoodchild commented:
Latest hotspot manager: 2c4af36f2b0b2ea3fe977f7044e85ea06e9ae1e5

Commit:
Reverted to old hotspot manager from 28156e074b2a2f498c3b1cffafb434b04a461080

chGoodchild commented:
DHCP works in commit 28156e074b2a2f498c3b1cffafb434b04a461080 , which is almost the same as commit 12c30aef294964ab6a6b8e3a52473f50967602b7 , but it doesn't work in 12c30aef294964ab6a6b8e3a52473f50967602b7... 

Will reset everything to 28156e074b2a2f498c3b1cffafb434b04a461080 and work forwards...
Latest hotspot manager: 2c4af36f2b0b2ea3fe977f7044e85ea06e9ae1e5
Latest first-boot-script, dhcp-lease-fix and setup_vpn: 12c30aef294964ab6a6b8e3a52473f50967602b7

Commit:
Reverted to state from 28156e074b2a2f498c3b1cffafb434b04a461080

chGoodchild commented:
Wow! 2ac6028708df070b1280d947b9b6add42fa16c6c doesn't work even though 28156e074b2a2f498c3b1cffafb434b04a461080 does. The only difference is that `dhcp_lease_fix.sh` is on the filesystem, but it isn't being used...

Commit:
Removed dhcp_lease_fix

Wow! 2ac6028 doesn't work even though 28156e0 does. The only difference is that dhcp_lease_fix.sh is on the filesystem, but it isn't being used...

chGoodchild commented:
According to diff there are differences between dcdc8096cf9c2d901de58c0e6c1702f7c6326ec9 and 28156e074b2a2f498c3b1cffafb434b04a461080 , but dcdc8096cf9c2d901de58c0e6c1702f7c6326ec9 doesn't give me a DHCP lease on first boot and 28156e074b2a2f498c3b1cffafb434b04a461080 does. 

Will build dcdc8096cf9c2d901de58c0e6c1702f7c6326ec9 from scratch to see what happens...

chGoodchild commented:
Built dcdc8096cf9c2d901de58c0e6c1702f7c6326ec9 from scratch and it still doesn't get a DHCP lease on firstboot. 

```
$ git log
commit dcdc8096cf9c2d901de58c0e6c1702f7c6326ec9 (HEAD -> 22-manage-wifi, origin/22-manage-wifi)
$ git diff 28156e074b2a2f498c3b1cffafb434b04a461080

```

No diff to 28156e074b2a2f498c3b1cffafb434b04a461080 !

chGoodchild commented:
It seems to depend on the order in which commits are built and whether they are built from scratch or not. I compiled 28156e074b2a2f498c3b1cffafb434b04a461080 after compiling dcdc8096cf9c2d901de58c0e6c1702f7c6326ec9 from scratch and 28156e074b2a2f498c3b1cffafb434b04a461080 no longer gives me a DHCP lease. However, earlier it did give me a DHCP lease every time I reverted to that commit and I compiled it (not from scratch though). 

I will compile 28156e074b2a2f498c3b1cffafb434b04a461080 from scratch now to verify that I don't get a DHCP lease in this commit and that it must have worked by coincidence due to something that was still left in the `openwrt` build directory from earlier commits (before 28156e074b2a2f498c3b1cffafb434b04a461080).

chGoodchild commented:
> It seems to depend on the order in which commits are built and whether they are built from scratch or not. I compiled [28156e0](https://github.com/OpenTollGate/TollGateNostrToolKit/commit/28156e074b2a2f498c3b1cffafb434b04a461080) after compiling [dcdc809](https://github.com/OpenTollGate/TollGateNostrToolKit/commit/dcdc8096cf9c2d901de58c0e6c1702f7c6326ec9) from scratch and [28156e0](https://github.com/OpenTollGate/TollGateNostrToolKit/commit/28156e074b2a2f498c3b1cffafb434b04a461080) no longer gives me a DHCP lease. However, earlier it did give me a DHCP lease every time I reverted to that commit and I compiled it (not from scratch though).
> 
> I will compile [28156e0](https://github.com/OpenTollGate/TollGateNostrToolKit/commit/28156e074b2a2f498c3b1cffafb434b04a461080) from scratch now to verify that I don't get a DHCP lease in this commit and that it must have worked by coincidence due to something that was still left in the `openwrt` build directory from earlier commits (before [28156e0](https://github.com/OpenTollGate/TollGateNostrToolKit/commit/28156e074b2a2f498c3b1cffafb434b04a461080)).
> 
> I'm re-flashing with `sysupgrade -n` by the way...

Confirmed, 28156e074b2a2f498c3b1cffafb434b04a461080 doesn't give me a DHCP lease after compiling from scratch...

Commit:
Reverted to 2c4af36f2b0b2ea3fe977f7044e85ea06e9ae1e5

Commit:
Removed make config

chGoodchild commented:
Double check before merge:

- [x] turn off wifi when `ndsctl json` isn't ready yet.
- [ ] Turn off wifi when `8.8.8.8` can't be pinged
- [x] use npub instead of MAC address if the npub is available.

Commit:
Why repeat if I anyway don't get a DHCP lease?

chGoodchild commented:
Postponing "use npub instead of MAC address if the npub is available" to a later feature, because this branch needs to be closed at some point and I don't want to test everything again from scratch..
